WordPress/Classes/ViewRelated/Me/All Domains/View Models/AllDomainsListViewModel.swift

```diff
@@ -0,0 +1,76 @@
+import Foundation
+import Combine
+
+class AllDomainsListViewModel {
+
+    // MARK: - Dependencies
+
+    private var domainsService: DomainsService?
+
+    // MARK: - Properties
+
+    @Published
+    private(set) var state: State = .empty
+
+    private var domains = [Domain]()
+
+    // MARK: - Init
+
+    init(coreDataStack: CoreDataStackSwift = ContextManager.shared) {
+        if let account = defaultAccount(with: coreDataStack) {
+            self.domainsService = .init(coreDataStack: coreDataStack, account: account)
+        }
+    }
+
+    private func defaultAccount(with contextManager: CoreDataStackSwift) -> WPAccount? {
+        try? WPAccount.lookupDefaultWordPressComAccount(in: contextManager.mainContext)
+    }
+
+    // MARK: - Load Domains
+
+    func loadData() {
+        self.state = .loading
+        self.fetchAllDomains { result in
+            switch result {
+            case .success(let domains):
+                self.domains = domains
+                self.state = .normal
```

Comment:
[Question] Do we need to check if the returned domains are empty before setting the state to `.normal`?

Reply:
If that's the case, I suggest removing this line and adding a didSet closure to domains

```swift
private var domains = [Domain]() {
    didSet {
        state = domains.isEmpty ? .empty : .normal
    }
}
```

Reply:
> [Question] Do we need to check if the returned domains are empty before setting the state to `.normal`?

That's correct, but I didn't focus too much on handling the empty state in this PR. I've opened another PR https://github.com/wordpress-mobile/WordPress-iOS/pull/21910implementing that feature.

Reply:
Makes sense 👍